src/main/java/frc/robot/subsystems/drivetrain/Drivetrain.java

```diff
@@ -0,0 +1,172 @@
+    rightFrontTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_RIGHT_FRONT_TALON);
+    leftMiddleTalon.follow(leftFrontTalon);
+    leftRearTalon.follow(leftFrontTalon);
+    rightMiddleTalon.follow(rightFrontTalon);
+    rightRearTalon.follow(rightFrontTalon);
+    drivetrain = new DifferentialDrive(leftFrontTalon, rightFrontTalon);
+    drivetrain.setDeadband(0);
+    // TODO: set correct port for pigeon gyro.
+    gyro = new Pigeon(robotConstants.can.DRIVETRAIN_LEFT_REAR_TALON);
+    kinematics = new DifferentialDriveKinematics(robotConstants.drivetrainConstants.WHEEL_BASE_WIDTH);
+    odometry = new DifferentialDriveOdometry(Rotation2d.fromDegrees(getAngle()));
+    // TODO: set correct talons for encoders.
+    leftEncoder = new WPI_TalonSRX(robotConstants.can.TEMPORARY_TALON_FOR_LEFT_DRIVETRAIN_ENCODER);
+    rightEncoder = new WPI_TalonSRX(robotConstants.can.TEMPORARY_TALON_FOR_RIGHT_DRIVETRAIN_ENCODER);
+  }
+
+  public void periodic() {
+    odometry.update(Rotation2d.fromDegrees(getAngle()), getLeftDistance(), getRightDistance());
+  }
+
+  // Drive functions
+  public void arcadeDrive(double x, double y) {
+    drivetrain.arcadeDrive(x, y, false);
+  }
+
+  public void curvatureDrive(double x, double y, boolean quickTurn) {
+    drivetrain.curvatureDrive(x, y, quickTurn);
+  }
+
+  public void tankDrive(double leftSpeed, double rightSpeed) {
+    drivetrain.tankDrive(leftSpeed, rightSpeed, false);
+  }
+
+  public void voltageTankDrive(double leftVoltage, double rightVoltage) {
+    tankDrive(leftVoltage / RobotController.getBatteryVoltage(), rightVoltage / RobotController.getBatteryVoltage());
+  }
+
+  public void move(double power) {
+    arcadeDrive(0, power);
+  }
+
+  // Gyro functions
+  double getAngle() {
+    return Math.IEEEremainder(gyro.getAngle(), 360);
+  }
+
+  public double getRadianAngle() {
+    return Math.toRadians(gyro.getAngle() );
+  }
+
+  public void resetGyro() {
+    gyro.reset();
+  }
+
+  public void calibrateGyro() {
+    gyro.calibrate();
+  }
+
+  // Encoders functions
+  public int getLeftTicks() {
+    return leftEncoder.getSelectedSensorPosition();
+  }
+
+  public int getRightTicks() {
+    return rightEncoder.getSelectedSensorPosition();
+  }
+
+  public void resetEncoders() {
+    leftEncoder.setSelectedSensorPosition(0);
+    rightEncoder.setSelectedSensorPosition(0);
+  }
+
+  public double getLeftDistance() {
+    return getLeftTicks() / robotConstants.drivetrainConstants.LEFT_ENCODER_TICKS_PER_METER;
+  }
+
+  public double getRightDistance() {
+    return getRightTicks() / robotConstants.drivetrainConstants.RIGHT_ENCODER_TICKS_PER_METER;
+  }
+
+  public double getAverageDistance() {
+    return (getLeftDistance() + getRightDistance()) / 2;
+  }
+
+  public double getRightVelocity() {
+    return rightEncoder.getSelectedSensorVelocity() / robotConstants.drivetrainConstants.RIGHT_ENCODER_TICKS_PER_METER
+        * 10;
+  }
```

Comment:
rightEncoder.getSelectedSensorVelocity() * 10 / robotConstants.drivetrainConstants.RIGHT_ENCODER_TICKS_PER_METER
also add doc that says the masured units